### DISCUSSION TITLE
Not sure why my autocompletion breaks.

### DISCUSSION THREAD

**Discussion by sylverek** [eclipse-langium/langium#1034] [Q&A] [answered]
To illustrate my issue I have replicated it with the basic Person/Greeting grammar. I extended the grammar as follows:

```
grammar HelloWorld

entry Model:
    (persons+=Person | greetings+=Greeting)*;

Person:
    'person' name=ID;

Greeting:
    'Hello' source=Ref 'from' target=Ref "\!";

Ref : Direct|Indirect;
Direct : "+" p=[Person];
Indirect : "\\" p=[Person];

hidden terminal WS: /\s+/;
terminal ID: /([_a-zA-Z][\w_]*)/;

hidden terminal ML_COMMENT: /\/\*[\s\S]*?\*\//;
hidden terminal SL_COMMENT: /\/\/[^\n\r]*/;
```

In summary, I modified ```Greeting``` and added ```Ref```, ```Direct```, and ```Indirect```.

My problem comes with the following document:
```
person John
person Jane

Hello +John from \!
```

When I place the cursor ```|``` in ```Hello +John from \|;``` , the autocompletion breaks and the following error is reported: ```Error: Could not find reference type for Greeting:p```

Now, if I add a space before the ```;``` like in ```Hello +John from \| ;```, the autocompletion works just fine and no error is reported. 

I do not understand why this is an issue. Did someone encounter something like that before? Anyone knows how to fix this?

A playground is available [here](https://langium.org/playground?grammar=OYJwhgthYgBAEgUwDbIPYHU0mQEwFD6IB2ALiAJ6wCyauKAXPrC7ABQAOiIAzmsTwDUAXgAK3PsVgAfWKESJSAS2LAhwgOIgFy1QEoAVAG5C43vyatYAci7ni12MUiJhASQAiJ-Fp0rglqzWSKhojnwAriAAxq4ASogAZjaJIGgQjqQwwIrCCckARAA6AIQF3vmwDLAeStrRpNJuxLh1iA0mtfWkVbAFggWwHMIA2maSALomza3dvcVFg8NjEvxThAAWSrj0UqTcECpgyLAYAMrVAPRFQpcm%2ByCHzieeV2wjAPpgALQAXgCC3wAWhMRkUAO4fCYGPR3TbbXawB5PY40AAyHwAwgB5ajUACiADkACpXIrXAxgnhFM7QgD8RQM5LhWx2JCRByOJzOGJxeKJpNg12FIwAekViEUQNC4fggA&content=A4UwTgzg9gdgBAKSgCxgKFJWiCGMRpoASIANqVHANRKpwBmYUAtnADoCEhQA)

Thanks!

**Comment by msujew** [comment, marked as answer]
Hey @sylverek,

you're encountering an issue I've outlined (and fixed) in https://github.com/langium/langium/pull/1004. The fix will get released next week (due to vacation).

**Reply by sylverek** [reply]
Thank you, @msujew . Really appreciate the work you guys are doing! Looking forward to next week. Enjoy your vacation!

**Reply by sylverek** [reply]
Hey @msujew ,

Is it possible that some changes after 1.2.0 brought that issue back?
I tried migrating to 1.3.0, 1.3.1 and even v2, but my autocompletion is breaking when I use something other than 1.2.0.

I get the same error as before, "Could not find reference type for ```Container.Type:property```" where the Container is the Container of the element after the cursor, and no completion result.

With the 1.2.0, I do catch the error as well, but I still get completion results, which I believe is because the algorithm looks before the cursor as well.

Any idea/recommendation?

Thank you.

**Reply by msujew** [reply]
Interesting. This might be related to the second part mentioned in https://github.com/eclipse-langium/langium/pull/1215. It's not a 100% fix, but we'll probably get there eventually.

No idea how to fix this on user side though, sorry. You can try the PR to see whether it fixes the issue for you.

Note that Langium 2.0 is effectively just 1.3 with the pretty breaking ESM change.

**Reply by cdietrich** [reply]
see also https://github.com/eclipse-langium/langium/issues/1216
